libs/contacts/ui/src/contacts/contact-dashboard.tsx

```diff
@@ -1,35 +1,82 @@
-import React, { Component } from 'react';
-import { Contact, ContactsService } from '@workshop/data-access';
+import React, { useEffect } from 'react';
+import { useLocation, Routes, Route } from 'react-router-dom';
+import { ContactsList } from './contact-list';
+import { useContactsHook } from './contacts.hooks';
+import { DependencyInjectionProvider } from '@mindspace/di';
+import { buildInjector } from '@workshop/data-access';
+const ContactDetailsWithId = React.lazy(() => import('./contact-details'));
 
-import { ContactDetails } from './contact-details';
+export const ContactDashboard = () => {
+  const location = useLocation();
+  const [people, doSearch] = useContactsHook();
 
-interface ContactsState {
-  people: Contact[];
-}
+  useEffect(() => {
+    doSearch('');
+  }, [doSearch, people]);
 
-export class ContactDashboard extends Component {
-  private service = new ContactsService();
+  return (
+    <DependencyInjectionProvider injector={buildInjector}>
```

Comment:
Why is this NOT a good idea? What are the consequences of defining the DI Provider here?